apps/web/app/modules/Courses/Lesson/Lesson.page.tsx

```diff
@@ -60,69 +60,54 @@ export default function LessonPage() {
   const lessonsIds = lessons.map((lesson) => lesson.id);
   const currentLessonIndex = lessonsIds.indexOf(lessonId ?? "");
 
-  if (!data || !lessonId) {
+  if (!lesson || !lessonId) {
     throw new Error(`Lesson with id: ${lessonId} not found`);
   }
 
-  const orderedLessonsItems = getOrderedLessons(data);
+  const orderedLessonsItems = getOrderedLessons(lesson);
 
   const questionsArray = getQuestionsArray(orderedLessonsItems);
 
   const previousLessonId = currentLessonIndex > 0 ? lessonsIds[currentLessonIndex - 1] : null;
   const nextLessonId =
     currentLessonIndex < lessonsIds.length - 1 ? lessonsIds[currentLessonIndex + 1] : null;
 
-  const isQuiz = data?.type === "quiz";
-  const isEnrolled = data?.enrolled;
+  const isQuiz = lesson?.type === "quiz";
+  const isEnrolled = lesson?.enrolled;
 
   const getScorePercentage = () => {
-    if (!data.quizScore || data.quizScore === 0 || !data.lessonItems.length) return "0%";
+    if (!lesson.quizScore || lesson.quizScore === 0 || !lesson.lessonItems.length) return "0%";
 
-    return `${((data.quizScore / data.lessonItems.length) * 100)?.toFixed(0)}%`;
+    return `${((lesson.quizScore / lesson.lessonItems.length) * 100)?.toFixed(0)}%`;
   };
 
   const scorePercentage = getScorePercentage();
 
-  const [lesson, setLesson] = useState(data);
-
-  const updateLessonItemCompletion = (lessonItemId: string) => {
-    setLesson((prevLesson) => {
-      const updatedLessonItems = prevLesson.lessonItems.map((item) => {
-        if (item.lessonItemId === lessonItemId && item.isCompleted === false) {
-          return { ...item, isCompleted: true };
-        }
-        return item;
-      });
-
-      return {
-        ...prevLesson,
-        lessonItems: updatedLessonItems,
-        itemsCompletedCount: updatedLessonItems.filter((item) => item.isCompleted).length ?? 0,
-      };
-    });
+  const updateLessonItemCompletion = async () => {
+    await refetch();
```

Comment:
If it is just a refetch, can we call it inside `LessonItem` without unnecessary props drilling? What do you think about this? Would this work?